Python/optimizer.c

```diff
@@ -324,6 +328,13 @@ add_to_pending_deletion_list(_PyExecutorObject *self)
 static void
 uop_dealloc(PyObject *op) {
     _PyExecutorObject *self = _PyExecutorObject_CAST(op);
+
+    // Object might be already untracked if we are in a GC cycle (via tp_clear).
+    // Avoid double-untracking.
+    if (_PyObject_GC_IS_TRACKED(self)) {
+        _PyObject_GC_UNTRACK(self);
+    }
+
```

Comment:
Remove these lines and try to see if it builds. This is causing a double untrack because we now untrack in invalidate.